runtime/doc/options.txt

```diff
@@ -5205,11 +5205,25 @@ A jump table for the options with a short description can be found at |Q_op|.
 	Note that such processing is done after |:set| did its own round of 
 	unescaping, so to keep yourself sane use |:let-&| like shown above.
 							*shell-powershell*
-	To use powershell: >
+	To use PowerShell: >
 		let &shell = has('win32') ? 'powershell' : 'pwsh'
-		set shellquote= shellpipe=\| shellxquote=
+		set shellquote= shellxquote=
 		set shellcmdflag=-NoLogo\ -NoProfile\ -ExecutionPolicy\ RemoteSigned\ -Command
-		set shellredir=\|\ Out-File\ -Encoding\ UTF8
+		set shellredir=2>&1\ \|\ Out-File\ -Encoding\ UTF8;\ exit\ $LastExitCode
+		set shellpipe=2>&1\ \|\ Out-File\ -Encoding\ UTF8;\ exit\ $LastExitCode
+
+< 	The above should work for all versions of PowerShell, but may result
+	in superfluous error object traces with PowerShell 5.1 due to how it
+	reacts when a running program writes anything to standard error stream.
+	This seems to be fixed in PowerShell versions 7.1 and above. The following
+	config may also be used to get a more Unix-like behaviour and echo make
+	output in the command window: >
+
+		let &shell = 'pwsh'
```

Comment:
Can you create a test for `pwsh` and skip it if it is unavailable?

Reply:
PTAL; I've split up pwsh bits into separate has_pwsh()/set_pwsh_shell()